### PR TITLE
Fix block deleted when settings are opened #1090

### PR DESCRIPTION
**Pull request by m-alzam**
Prevent block removal when settings are opened #1090

### CODE REVIEW

**Comment by neSpecc** [comment]
I think it would better to leave an ability to delete blocks this way. So it's better to close Block Settings in such a case.

**Comment by m-alzam** [comment]
> I think it would better to leave an ability to delete blocks this way. So it's better to close Block Settings in such a case.

But there's a problem with input elements inside settings. If you press the backspace key while editing text in a settings' input, the block is deleted instead of doing the default behavior (remove a character).

**Comment by neSpecc** [comment]
> > I think it would better to leave an ability to delete blocks this way. So it's better to close Block Settings in such a case.
> 
> But there's a problem with input elements inside settings. If you press the backspace key while editing text in a settings' input, the block is deleted instead of doing the default behavior (remove a character).

Ok, got this. We need to think about how to detect if the Caret inside some kind of an editable element at the settings. Or maybe allow an ability to prevent default behavior from the plugin side.

**Comment by gohabereg** [comment]
I think we can use `document.activeElement` for that

**Comment by m-alzam** [comment]
Thanks @neSpecc and @gohabereg for the suggestions. I'l try to implement the check if we're in an editable element tomorrow.

**Comment by m-alzam** [comment]
PR updated.

File: src/components/modules/ui.ts [hunk 1]
```diff
@@ -390,13 +390,23 @@ export default class UI extends Module {
   private backspacePressed(event: KeyboardEvent): void {
     const {BlockManager, BlockSelection, Caret} = this.Editor;
 
+    /** Prevent block removal while editing an input inside settings */
+    if (this.Editor.BlockSettings.opened) {
+      if (document.activeElement.tagName === 'INPUT') {
```
Comment (neSpecc) [inline src/components/modules/ui.ts:395]: we have a useful method for that, see `dom.canSetCaret()`

Comment (m-alzam) [reply src/components/modules/ui.ts:395]: Ok, thanks. It has more features. I'll update the PR this evening of I have time.

**Comment by neSpecc** [comment]
The solution works, but it only allows to press backspace at the input. There are also some problems with Flipper - class responsible for tab/arrow navigation between buttons:
- you can't navigate by arrows inside the `input` at the block tunes
- you can't focus the `input` with Flipper
- the input has no auto focus when block tunes are opened

**Comment by m-alzam** [comment]
Closing in favor of #1123

**Pull request closed without merging**